text/0496-handlebars-strict-mode.md

```diff
@@ -89,7 +100,47 @@ All of these problems are all related to implicit globals and/or implicit
 `this` fallback. Since neither of these features are supported in strict mode,
 they are no longer a concern for us.
 
-### 3. No dynamic resolution
+### 3. No implicit invocation of argument-less helpers
+
+In the contextual helpers RFC, we discussed [an issue](https://github.com/emberjs/rfcs/blob/master/text/0432-contextual-helpers.md#relationship-with-globals)
+regarding the invocation of arguments-less helpers in invocation argument
+positions.
+
+In today's semantics, if there is a global helper named `pi`, the following
+template will result it the helper being invoked, and the result of the helper
+invocation will be passed into the component.
+
+```hbs
+<MyComponent @value={{pi}} />
+```
+
+This is not desirable behavior in the value-based semantics proposed in that
+RFC, because it makes it impossible to pass helpers around as values, just as
+it is possible to pass around contextual components today.
+
+The contextaul helper RFC proposed to deprecate this behavior and require
+mandatory parentheses to invoke the helper.
+
+
+```hbs
+{{!-- passing the helper pi to the component --}}
+<MyComponent @value={{pi}} />
+
+{{!-- passing the result of invoking the helper pi to the component --}}
+<MyComponent @value={{(pi)}} />
+```
+
+In strict mode, the current, soon-to-be-deprecated behavior will be removed,
+and the parentheses syntax will be mandatory.
+
+Note that this only affects arguments-less helpers, which are exceedingly rare,
+as most helpers perform self-contained computations based on the provided
+arguments. It also only affect argument positions. In content and attribute
```

Comment:
> It also only affect argument positions.

This is _vaguely_ concerning to me. I think it may be somewhat difficult to teach...